test/verify/check-metrics

```diff
@@ -54,25 +54,25 @@ class TestMetrics(MachineCase):
         b = self.browser
         m = self.machine
 
+        # When checking whether the plots show the expected results,
+        # we look for a segment of the data of a certain duration
+        # whose average is in a certain range.  Otherwise any short
+        # outlier will make us miss the plateau.  Such outliers happen
+        # frequently with the CPU plot.
+
         b.eval_js("""
           ph_plateau = function (data, min, max, duration, label) {
-            var i, start = undefined, start_index, sum;
+            var i, j, sum, avg;
             for (i = 0; i < data.length; i++) {
-              if ((min === null || data[i][1] >= min) && (max === null || data[i][1] <= max)) {
-                if (start === undefined) {
-                  start = data[i][0];
-                  start_index = i;
-                  sum = 0;
-                }
-                sum += data[i][1];
-                if (data[i][0] - start >= duration * 1000) {
-                  console.log("avg", label, sum/(i - start_index + 1));
-                  return true;
+              sum = 0;
+              for (j = i; j < data.length; j++) {
```

Comment:
Yes, it's O(n^2), but does it matter?  (I know these things come back to hunt one later...)

Actually, the j loop needs to end whenever the window is large enough.